File: ropeCuttingQuestion/ropeQuestion.php
<?php
    function cut_rope($rope_length)
    {
        $rope_threes = floor($rope_length / 3);
        $rope_threes_remainder = $rope_length % 3;
        $rope_twos = 0;

        switch($rope_threes_remainder)
        {
            case 1:
                $rope_threes--;
                $rope_twos = 2;
                break;
            case 2:
                $rope_twos = 1;
                break;
        }

        return (3 ** $rope_threes) * (2 ** $rope_twos);
    }

    echo cut_rope(8);
    echo "\n";
    echo cut_rope(9);
    echo "\n";
    echo cut_rope(10);
?>

File: ropeCuttingQuestion/ropeQuestion.java
class ropeQuestion
{
    public static Integer CutRope(int ropeLength)
    {
        int ropeThrees = 0;
        int ropeThreesRem = 0;
        int ropeTwos = 0;

        ropeThrees = ropeLength / 3;
        ropeThreesRem = ropeLength % 3;


        switch(ropeThreesRem)
        {
            case 1:
                ropeThrees--;
                ropeTwos = 2;
                break;
            case 2:
                ropeTwos = 1;
                break;
        }

        return (int) (Math.pow(3, ropeThrees) * Math.pow(2, ropeTwos));
    }


    public static void main(String[] args)
    {
        System.out.println(CutRope(8));
        System.out.println(CutRope(9));
        System.out.println(CutRope(10));
    }
}